### PR TITLE
Initial implementation of the MIKE.NET module

### PR DESCRIPTION
**Pull request by gedaskir**
MIKE.NET is a module which creates helper methods for loading and referencing MIKE software .NET libraries. This also allows to use MIKE installation libraries different from the ones shipped by MIKE IO 1D.

### CODE REVIEW

File: mikeio1d/mikenet/library_loader.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        A method which adds a reference of a relevant library to Python.NET (CLR)
+        and also adds a reference to the MIKE.NET.
+        """
+        clr.AddReference(self.library_name)
+
+        try:
+            mikenet_dict = self.mikenet_module.__dict__
+            exec(f"import {self.library_name} as {self.library_alias}", mikenet_dict)
+        except:
+            pass
```
[the remark this answers is not in the document]
Comment (gedaskir) [reply mikeio1d/mikenet/library_loader.py]: I added a warning here.

File: mikeio1d/mikenet/library_loaders.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    mikenet_module: module
+        Reference to MIKE.NET module
+    """
+
+    def __init__(self, mikenet_module):
+        self.mikenet_module = mikenet_module
+        self.library_loader_list = []
+        self.library_loader_dict = {}
+
+        setup_mike_installation(sys.path)
```
[the remark this answers is not in the document]
Comment (gedaskir) [reply mikeio1d/mikenet/library_loaders.py]: This is a good idea. I wrapped the variables in mikepath.py into MikePath class.

File: mikeio1d/mikenet/library_loaders.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        self.create_loaders()
+
+    def create_loaders(self):
+        """
+        Creates .NET library loaders and assigns relevant `load` methods
+        to MIKE.NET module.
+        """
+        mikenet_module = self.mikenet_module
+
+        for pattern in library_patterns:
```
[the remark this answers is not in the document]
Comment (gedaskir) [reply mikeio1d/mikenet/library_loaders.py]: Same as above.

File: mikeio1d/mikepath.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        from DHI.Mike.Install import MikeImport
+
+        MikeImport.SetupInstallDir(mike_install_path)
+
+
+mikeio1d_bin_path = os.path.join(os.path.dirname(__file__), "bin")
+
+mike_install_path = os.environ.get("MIKE_INSTALL_PATH", None)
+
+skip_bin_x64 = os.environ.get("SKIP_BIN_X64", None)
```
[the remark this answers is not in the document]
Comment (gedaskir) [reply mikeio1d/mikepath.py]: Agree :-).

File: mikeio1d/mikenet/library_loader.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    def load(self):
+        """
+        A method which adds a reference of a relevant library to Python.NET (CLR)
+        and also adds a reference to the MIKE.NET.
+        """
+        clr.AddReference(self.library_name)
+
+        try:
+            mikenet_dict = self.mikenet_module.__dict__
+            exec(f"import {self.library_name} as {self.library_alias}", mikenet_dict)
```
[the remark this answers is not in the document]
Comment (ecomodeller) [reply mikeio1d/mikenet/library_loader.py:40]: How about using [`importlib.import_module`](https://docs.python.org/3/library/importlib.html#importlib.import_module)?

**Pull request merged**